### PR TITLE
arrow length based on simulation bounds

### PR DESCRIPTION
**Pull request by tylerflex**
source and monitor plotting accepts an optional `sim_bounds` argument.
If supplied, the arrow lengths are determined by the ARROW_LENGTH_FACTOR times the minimum length or width between the bounds on the plotting plane.

This avoids issues with using the axis limits, which might be temporarily resized for infinite monitor or source object, such as plane waves.

### CODE REVIEW

**Review by shashwat-sh — APPROVED** [review]
I think it's a good solution. The idea of providing `bounds` works, but to a user it may not be obvious that the solution to this issue is to supply `bounds`. Maybe we can add an example of this in one of the visualization notebooks?

I'm wondering if there's a way to preempt the situation in an imperfect way but one that still works reasonably in the event that the user does not supply `bounds`. E.g., maybe there could be a default axis size (width, height) specified in viz.py (counting on the fact that most python plots will be in the ballpark of ~400 x ~400 pixels, for example?) and then using that to compute arrow length if bounds are not supplied AND size is inf.

I don't think the above is critical though.

**Comment by tylerflex** [comment]
I think the only time this is an issue is if the following conditions are met:
1. The source (or monitor) has infinite (or very large) bounds.
2. The source (or monitor) is plotted from within a `Simulation.plot_xxx()` function.

If the user tries to do `plane_wave.plot(z=0)`, it will just display the entire thing on an axis with bounds extending the full length of the (almost) infinite object. So it works without providing sim_bounds.

Basically, the way it's set up now, the user should never really use sim_bounds, it's more a way for us to tag the arrow plotting as happening within another plotting context and make sure that the correct size is specified.

**Comment by shashwat-sh** [comment]
Ok, sounds good then, I don't have any more comments.

**Comment by tylerflex** [comment]
ok, thanks for looking it over
